pandas/tests/io/test_excel.py

```diff
@@ -1058,51 +1008,34 @@ def test_read_excel_squeeze(self):
         tm.assert_series_equal(actual, expected)
 
 
-class TestXlsReaderTests(XlrdTests):
-    ext = '.xls'
-    engine_name = 'xlrd'
-    check_skip = staticmethod(_skip_if_no_xlrd)
-
-
-class TestXlsxReaderTests(XlrdTests):
-    ext = '.xlsx'
-    engine_name = 'xlrd'
-    check_skip = staticmethod(_skip_if_no_xlrd)
+class _WriterBase(SharedItems):
+    """Provides fixture to set / reset options for all writer tests"""
 
+    @pytest.fixture(autouse=True)
+    def set_options(self, request, merge_cells, engine, ext):
+        option_name = 'io.excel.{ext}.writer'.format(ext=ext.strip('.'))
+        prev_engine = get_option(option_name)
+        set_option(option_name, engine)
+        yield
+        set_option(option_name, prev_engine)
 
-class TestXlsmReaderTests(XlrdTests):
-    ext = '.xlsm'
-    engine_name = 'xlrd'
-    check_skip = staticmethod(_skip_if_no_xlrd)
 
-
-class ExcelWriterBase(SharedItems):
+@pytest.mark.parametrize("merge_cells", [True, False])
+@pytest.mark.parametrize("engine,ext", [
+    pytest.param('openpyxl', '.xlsx', marks=pytest.mark.skipif(
+        not td.safe_import('openpyxl'), reason='No openpyxl')),
+    pytest.param('xlwt', '.xls', marks=pytest.mark.skipif(
+        not td.safe_import('xlwt'), reason='No xlwt')),
+    pytest.param('xlsxwriter', '.xlsx', marks=pytest.mark.skipif(
+        not td.safe_import('xlsxwriter'), reason='No xlsxwriter'))
+])
+class TestExcelWriter(_WriterBase):
```

Comment:
Rather than creating subclasses and adding arbitrary class metadata it seemed easy enough and clearer to parametrize the writing tests for each engine and the extension it should be serving